ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/PivotBlockConfirmer.java

```diff
@@ -187,9 +187,14 @@ private CompletableFuture<BlockHeader> executePivotQuery(final long blockNumber)
       LOG.debug("No peer currently available to query for block {}.", blockNumber);
       pivotHeaderFuture =
           ethContext
-              .getScheduler()
-              .timeout(WaitForPeerTask.create(ethContext, metricsSystem), Duration.ofSeconds(5))
-              .handle((err, res) -> null) // Ignore result
+              .getEthPeers()
+              .waitForPeer(
+                  (peer) ->
+                      peer.chainState().getEstimatedHeight() >= blockNumber
```

Comment:
Similar to the previous comment in FastSyncActions, the pivot block is still being used in the POS syncing and we can't rely on the estimated height. So should either get rid of that or guard with a isPOS check here.

Reply:
I think just removing it should be fine. If the peer isn't suitable when we try to use it, we'll end up back here again eventually anyway.